cli/ballerina-launcher/src/main/java/org/ballerinalang/launcher/util/BCompileUtil.java

```diff
@@ -280,8 +291,20 @@ private static CompileResult compile(CompilerContext context, String packageName
         } else if (CompilerPhase.CODE_GEN.compareTo(compilerPhase) > 0 || compilerPhase == CompilerPhase.BIR_GEN) {
             return comResult;
         }
+        CompiledBinaryFile.ProgramFile programFile;
+        // Check if its executing the tests. If not then it is used to compile a ballerina source file, then pass the
+        // bLangPackage node to generate the program file
+        if (!withTests) {
```

Comment:
We dont need three branches here. It can be merged to an if and else. Pl check the logic.